store/errors.go

```diff
@@ -137,6 +137,42 @@ type SnapActionError struct {
 	Other []error
 }
 
+// SingleOpError returns the single operation, snap name, and error if
+// e represents a single error of a single operation on a single snap
+// (i.e. if e.Other is empty, and e.Refresh, e.Install and e.Download
+// have a single error in total).
+// In any other case, the error returned will be nil.
+func (e SnapActionError) SingleOpError() (op, name string, err error) {
+	if len(e.Other) > 0 {
+		return "", "", nil
+	}
+
+	nRefresh := len(e.Refresh)
+	nInstall := len(e.Install)
+	nDownload := len(e.Download)
+	if nRefresh+nInstall+nDownload != 1 {
+		return "", "", nil
+	}
+
+	var errs map[string]error
+	switch {
+	case nRefresh > 0:
+		op = "refresh"
+		errs = e.Refresh
+	case nInstall > 0:
+		op = "install"
+		errs = e.Install
+	case nDownload > 0:
+		op = "download"
+		errs = e.Download
```

Comment:
Should we have a "default" here that does something if we ever add a new action and forget to update this switch ?

Reply:
because of the if before we cannot get in that situation, this would simply return ""  above already